cpp/FbxCliWrapper/FbxCliWriter.cpp

```diff
@@ -0,0 +1,188 @@
+		}
+
+		// Destroy the exporter.
+		lExporter->Destroy();
+
+		return result;
+	}
+
+	void FbxCliWriter::ExportNodes()
+	{
+		// Create all the FBX meshes and nodes
+		std::vector<FbxMesh*> meshList;
+		CreateFBXMeshList(meshList);
+		CreateFBXNodes(meshList);
+	}
+
+	void FbxCliWriter::CreateFBXNodes(std::vector<FbxMesh*>& MeshList)
+	{
+		// Create all the fbx nodes required
+		// Skip the IScene's root node
+		FbxNode* lRootNode = mScene->GetRootNode();
+		std::vector<FbxNode*> nodeList;
+		for (int nodeIndex = 1; nodeIndex < (int)mSceneData->mNodeNameList.size(); nodeIndex++)
+		{
+			auto nodeName = mSceneData->mNodeNameList[nodeIndex];
+
+			FbxNode* newNode = FbxNode::Create(mScene, nodeName.c_str());
+			nodeList.push_back(newNode);
+		}
+
+		// use the parent index list to insert children into correct nodes
+		for (int nodeIndex = 0; nodeIndex < mSceneData->mNodeParentList.size(); nodeIndex++)
+		{
+			int32_t meshIndex = mSceneData->mNodeMeshIndexList[nodeIndex];
+			int32_t nodeParentIndex = mSceneData->mNodeParentList[nodeIndex];
+			FbxMatrix nodeLocalTransform;
+			auto node = nodeIndex == 0 ? lRootNode : nodeList[nodeIndex - 1];
+
+			if (nodeParentIndex == -1)
+			{
+				nodeLocalTransform = *reinterpret_cast<FbxMatrix*>(&mSceneData->mNodeTransformList[nodeIndex]);
+			}
+			else if (nodeParentIndex == 0)
+			{
+				auto nodeGlobalTransform = *reinterpret_cast<FbxMatrix*>(&mSceneData->mNodeTransformList[nodeIndex]);
+				auto parentGlobalTransform = *reinterpret_cast<FbxMatrix*>(&mSceneData->mNodeTransformList[nodeParentIndex]);
+				nodeLocalTransform = parentGlobalTransform.Inverse() * nodeGlobalTransform;
+				//nodeLocalTransform = *reinterpret_cast<FbxMatrix*>(&mSceneData->mNodeTransformList[nodeIndex]);
+				bool res = lRootNode->AddChild(node);
+			}
+			else
+			{
+				auto nodeGlobalTransform = *reinterpret_cast<FbxMatrix*>(&mSceneData->mNodeTransformList[nodeIndex]);
+				auto parentGlobalTransform = *reinterpret_cast<FbxMatrix*>(&mSceneData->mNodeTransformList[nodeParentIndex]);
+				nodeLocalTransform = parentGlobalTransform.Inverse() * nodeGlobalTransform;
+
+				bool res = nodeList[nodeParentIndex - 1]->AddChild(node);
+			}
+
+			if (meshIndex != -1)
+			{
+				// Set the node attribute of the mesh node.
+				node->SetNodeAttribute(MeshList[meshIndex]);
+			}
+
+			FbxVector4 pTranslation;
+			FbxVector4 pRotation;
+			FbxVector4 pShearing;
+			FbxVector4 pScaling;
+			double pSign;
+			nodeLocalTransform.GetElements(pTranslation, pRotation, pShearing, pScaling, pSign);
+
+			node->LclTranslation = pTranslation;
+			node->LclRotation = pRotation;
+			node->LclScaling = pScaling;
+		}
+	}
+
+	void FbxCliWriter::CreateFBXMeshList(std::vector<FbxMesh*> & MeshList)
+	{
+		// Create all the FBX meshes
+		for (int meshIndex = 0; meshIndex < (int)mSceneData->mMeshList.size(); meshIndex++)
+		{
+			// Create a mesh.
+			FbxMesh* lMesh = FbxMesh::Create(mScene, mSceneData->mMeshIdList[meshIndex].c_str());

```

Comment:
`auto` is sad, because he isn't being used

Reply:
lol